esp-hal/src/uart.rs

```diff
@@ -400,12 +403,52 @@ pub struct Config {
     pub stop_bits: StopBits,
     /// Clock source used by the UART peripheral.
     pub clock_source: ClockSource,
+    /// UART Recieve part configuration.
+    pub rx: RxConfig,
+    /// UART Transmit part configuration.
+    pub tx: TxConfig,
+}
+
+/// UART Recieve part configuration.
+#[derive(Debug, Clone, Copy, procmacros::BuilderLite)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[non_exhaustive]
+pub struct RxConfig {
     /// Threshold level at which the RX FIFO is considered full.
     pub rx_fifo_full_threshold: u16,
     /// Optional timeout value for RX operations.
     pub rx_timeout: Option<u8>,
```

Comment:
```suggestion
    pub fifo_full_threshold: u16,
    /// Optional timeout value for RX operations.
    pub timeout: Option<u8>,
```

Sorry to nitpick, but we already know these are for RX, so these prefixes are redundant.